src/modules/RootPart.rb

```diff
@@ -2209,22 +2206,78 @@ def update_staging_partitions!(activated_partitions, mountpoint = nil)
     def update_staging_filesystem!(name, mountpoint)
       log.info "Setting partition data: Device: #{name}, MountPoint: #{mountpoint}"
 
+      # storage-ng
+      #
+      # FIXME
+      #
+      # The code below does not work as one would expect as 'name' comes straight out of
+      # /etc/fstab and might look like 'UUID=2f61fdb9-f82a-4052-8610-1eb090b82098'.
+      #
+      # The only value of this seems to be for yast-bootloader to locate the
+      # root & boot devices.
+      #
+      # Note that this works magically atm because for the root device, the
+      # kernel device name is passed so the 'if' below actually matches.
+      #
+
       if name.include?("/dev/disk/by-id")
         mount_by = Y2Storage::Filesystems::MountByType::ID
-        filesystem = staging.partitions.with(id: name).filesystems.first
       elsif name.include?("/dev/")
         mount_by = Y2Storage::Filesystems::MountByType::DEVICE
-        filesystem = staging.partitions.with(name: name).filesystems.first
       else
-        mount_by = Y2Storage::Filesystems::MountByType::LABEL
-        filesystem = staging.filesystems.with(label: name).first
+        # existing code has the following lines here that don't make sense (afaics):
+        #
+        # mount_by = Y2Storage::Filesystems::MountByType::LABEL
+        # filesystem = staging.filesystems.with(label: name).first
       end
 
-      return unless filesystem
+      return unless mount_by
+
+      filesystem = fs_by_devicename(staging, name)
+
+      if filesystem
+        filesystem.mountpoint = mountpoint
+        filesystem.mount_by = mount_by
+      end
 
-      filesystem.add_mountpoint(mountpoint)
-      filesystem.mount_by = mount_by.to_i
     end
+
+    # Look up filesystem type for a device.
+    #
+    # Return nil if there's no such device or device doesn't have a filesystem.
+    #
+    # @param devicegraph [Devicegraph]
+    # @param devicename [String]
+    # @return [String, nil]
+    #
+    def fstype_for_device(devicegraph, devicename)
+      fs = fs_by_devicename(devicegraph, devicename)
+      fs.type.to_s if fs
+    end
+
+    # Look up filesystem object with matching device name.
+    #
+    # Return nil if there's no such device or the device doesn't have a
+    # filesystem.
+    #
+    # @param devicegraph [Devicegraph]
+    # @param devicename [String]
+    # @return [Y2Storage::Filesystems::BlkFilesystem, nil]
+    #
+    def fs_by_devicename(devicegraph, devicename)
+      fs = devicegraph.blk_filesystems.find do |fs|
+        fs.blk_devices.any? { |dev| dev.name == devicename }
+      end
+
+      # log a bit
```

Comment:
I wanted to see in the log on which devicegraph we operate. The method is used both on `probed`
and `staging` while the relevant parts seem to happen in `staging` (which in theory is just a copy of `probed` anyway).

In general I have many doubts about the code in `RootPart.rb` and suspect that quite a large part of it is dysfunctional or outright broken. But I feel that addressing this goes beyond this pbi.